idioms/ffi-passing-strings.md

```diff
@@ -0,0 +1,74 @@
+# Passing Strings
+
+## Description
+
+When passing strings to FFI functions, there are four principles that should be followed:
+
+1. Make the lifetime of owned strings as long as possible.
+1. Minimize `unsafe` code during the conversion.
+1. If the C code can modify the string data, use `Vec` instead of `CString`.
+1. Unless the FFAPI requires it, the ownership of the string should not transfer to the callee.
+
+## Motivation
+
+Rust has built-in support for C-style strings with its `CString` and `CStr` types. However, there are different approaches one can take with strings that are being sent to a foreign function call from a Rust function.
+
+The best practice is simple: use `CString` in such a way as to minimize unsafe code. However, a secondary caveat is that *the object must live long enough*, meaning the lifetime should be maximized. In addition, the documentation explains that "round-tripping" a `CString` after modification is UB, so additional work is necessary in that case.
```

Comment:
Will be easier for changes to follow due to VCS caveats. Also I propose to use `unsafe` instead of unsafe in the complete PR to make sure that it stands out to the reader.

```suggestion
The best practice is simple: use `CString` in such a way as to minimize `unsafe` code. 
However, a secondary caveat is that *the object must live long enough*, meaning the lifetime should be maximized.
In addition, the documentation explains that "round-tripping" a `CString` after modification is UB, so additional work is necessary in that case.
```